core/source.ts

```diff
@@ -36,8 +38,20 @@ export default class SiteSource implements Source {
     site.addEventListener("beforeUpdate", (ev: Event) => {
       this.root.refreshCache();
 
-      for (const filename of ev.files!) {
+      // copy files list so the following loop is not affected by mutation
+      const files = [...ev.files!];
+
+      for (const filename of files) {
         this.#cache.delete(site.src(filename));
+
+        const dependants = this.#dependants.get(filename);
+
+        if (dependants) {
+          dependants.forEach((dependant) => {
+            ev.files!.add(dependant);
+            this.#cache.delete(site.src(dependant));
+          });
+        }
```

Comment:
If the current module has dependants, we force those files to be included in the update.